### PR TITLE
Create Performing the Run process.md

### PR DESCRIPTION
**Pull request by LiangWenshuo1118**
Description of the steps and output files in the DP-GEN run process.

### CODE REVIEW

**Comment by AnguseZhang** [comment]
I've provided my suggestions with @LiangWenshuo1118.

File: doc/run/Overview-of-the-run-process.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+02.fp : Selected structures will be calculated by first principles methods(default VASP). DP-GEN will obtain some new data and put them together with initial data and data generated in previous iterations. After that a new training will be set up and DP-GEN will enter next iteration!
+
+In the run process of the DP-GEN, we need to specify the basic information about the system, the initial data, details of the training, exploration, and labeling tasks. In addition, we need to specify the software, machine environment, and computing resource and enable the process of job generation, submission, query, and collection automatically. We can perform the run process as we expect by specifying the keywords in param.json and machine.json, and they will be introduced in detail in the following sections. 
+
+Here, we give a general description of the run process. We can execute the run process of DP-GEN easily by:
+
+```
```
Comment (njzjz) [inline doc/run/Overview-of-the-run-process.md:15]:
```suggestion
```sh
```
See https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting

File: doc/run/example-of-param.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+# Example-of-param.json
+
+We have provided different examples of param.json in dpgen/examples/run/. In this section, we give a description of the param.json, taking dpgen/examples/run/dp2.x-lammps-vasp/param_CH4_deepmd-kit-2.0.1.json as an example. This is a param.json for a gas-phase methane molecule. Here, DeePMD-kit (v2.x), LAMMPS and VASP codes are used for training, exploration and labelling respectively.
+
+## basics
+
+The basics related keys in param.json is given as follows
+
+```
```
Comment (njzjz) [inline doc/run/example-of-param.md:9]:
```suggestion
```json
```

**Review by njzjz — CHANGES_REQUESTED** [review]

File: doc/run/example-of-machine.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        "group_size": 1,
+        "source_list": ["/home/user1234/deepmd.env"]
+      }
+    }
+  ],
+```
+
+The "command" for the train task in the DeePMD-kit is "dp".
+
+In machine parameters, "batch_type" specifies the type of job scheduling system. If there is no job scheduling system, we can use the "Shell" to perform the task. "context_type" specifies the method of data transfer, and "local" means copying and moving data via local file storage systems (e.g. cp, mv, etc.). In DP-GEN, the paths of all tasks are automatically located and set by the software, and therefore "local_root" is always set to ". /". The input file for each task will be sent to the "remote_root" and the task will be performed there, so we need to make sure that the path exists.
```
Comment (njzjz) [inline doc/run/example-of-machine.md:45]:
```suggestion
In machine parameters, "batch_type" specifies the type of job scheduling system. If there is no job scheduling system, we can use the "Shell" to perform the task. "context_type" specifies the method of data transfer, and "local" means copying and moving data via local file storage systems (e.g. cp, mv, etc.). In DP-GEN, the paths of all tasks are automatically located and set by the software, and therefore "local_root" is always set to "./". The input file for each task will be sent to the "remote_root" and the task will be performed there, so we need to make sure that the path exists.
```

`. /` -> `./` (remove space)

File: doc/run/example-of-machine.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        "group_size": 5,
+        "source_list": ["/home/user1234/vasp.env"]
+      }
+    }
+],
+```
+
+VASP code is used for fp task and mpi is used for parallel computing, so "mpirun -n 32" is added to specify the number of parallel threads.
+
+In the machine parameter, "context_type" is modified to "SSHContext" and "batch_type" is modified to "PBS". It is worth noting that "remote_root" should be set to an accessible path on the remote PBS cluster. "remote_profile" is added to specify the information used to connect the remote cluster, including hostname, username, password, port, etc.
```
Comment (njzjz) [inline doc/run/example-of-machine.md:114]:
```suggestion
In the machine parameter, "context_type" is modified to "SSHContext" and "batch_type" is modified to "PBS". It is worth noting that "remote_root" should be set to an accessible path on the remote PBS cluster. "remote_profile" is added to specify the information used to connect the remote cluster, including hostname, username, port, etc. 
```
Password should be discouraged. Instead, we should encourage SSH keys in the doc.

File: doc/run/example-of-param.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+```json
+  "numb_models": 4,
+  "train_param": "input.json",
+  "default_training_param": {
+  },
+```
+The training related keys specify the detalis of training tasks. "numb_models" specifies the number of models to be trained. "default_training_param" specify the training parameters for `deepmd-kit`. 
+
+Here, 4 DP models will be trained in `00.train`. Detailed explanation of training parameters can be found at [DeePMD-kit’s documentation](https://docs.deepmodeling.com/projects/deepmd/en/master/).
```
Comment (njzjz) [inline doc/run/example-of-param.md:59]: Provide the specific page instead.

File: doc/run/overview-of-the-run-process.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+| 0                    | 3                           | make_model_devi  |
+| 0                    | 4                           | run_model_devi   |
+| 0                    | 5                           | post_model_devi  |
+| 0                    | 6                           | make_fp          |
+| 0                    | 7                           | run_fp           |
+| 0                    | 8                           | post_fp          |
+
+0,1,2 correspond to make_train, run_train, post_train. DP-GEN will write scripts in make_train, run the task by specific machine in run_train and collect result in post_train. The records for model_devi and fp stage follow similar rules.
+
+If the process of DP-GEN stops for some reason, DP-GEN will automatically recover the main process by record.dpgen. You may also change it manually for your purpose, such as removing the last iterations and recovering from one checkpoint.
```
Comment (njzjz) [inline doc/run/overview-of-the-run-process.md:65]:
```suggestion
If the process of DP-GEN stops for some reasons, DP-GEN will automatically recover the main process by record.dpgen. You may also change it manually for your purpose, such as removing the last iterations and recovering from one checkpoint.
```

**Review by njzjz — COMMENTED** [review]
Can you add these doc to `doc/index.rst`?

**Comment by njzjz** [comment]
1. For maintenance, it's better to submit the original script instead of the generated doc.
2. It's better to separate it into a new PR instead of doing all things in a single PR.

**Review by njzjz — APPROVED** [review]

**Review by AnguseZhang — APPROVED** [review]

**Pull request merged**